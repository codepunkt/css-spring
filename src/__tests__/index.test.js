import spring from '../index'

describe('css-spring', () => {
  test('matches snapshots', () => {
    expect(
      spring(
        {
          opacity: 0,
        }, {
          opacity: 1,
        })
    ).toMatchSnapshot()

    expect(
      spring(
        {
          'padding-left': '-50px',
          background: '#bada55',
        },
        {
          'padding-left': '50px',
          background: '#f00',
        },
        {
          preset: 'wobbly',
          precision: 5,
        }
      )
    ).toMatchSnapshot()

    expect(
      spring(
        {
          'padding-left': '-50rem',
          opacity: true,
          'margin-right': '0em',
          border: '1px solid #f00',
          transform: 'translateX(0) translateY(0) scaleX(1) scaleY(1)',
        },
        {
          'padding-left': '50px',
          opacity: 1,
          'margin-right': '127em',
          border: '10px solid #f00',
          transform: 'translateX(100px) translateY(100px) scaleX(2) scaleY(2)',
        },
        {
          preset: 'gentle',
          damping: 50,
          stiffness: 80,
        }
      )
    ).toMatchSnapshot()
  })
})
