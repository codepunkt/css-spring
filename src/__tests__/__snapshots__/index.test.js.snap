exports[`css-spring matches snapshots 1`] = `
Object {
  "0%": Object {
    "opacity": 0.02,
  },
  "1%": Object {
    "opacity": 0.05,
  },
  "10%": Object {
    "opacity": 0.68,
  },
  "100%": Object {
    "opacity": 1,
  },
  "11%": Object {
    "opacity": 0.75,
  },
  "12%": Object {
    "opacity": 0.82,
  },
  "13%": Object {
    "opacity": 0.88,
  },
  "14%": Object {
    "opacity": 0.94,
  },
  "15%": Object {
    "opacity": 0.99,
  },
  "16%": Object {
    "opacity": 1.04,
  },
  "17%": Object {
    "opacity": 1.08,
  },
  "18%": Object {
    "opacity": 1.11,
  },
  "19%": Object {
    "opacity": 1.14,
  },
  "2%": Object {
    "opacity": 0.1,
  },
  "20%": Object {
    "opacity": 1.16,
  },
  "21%": Object {
    "opacity": 1.18,
  },
  "22%": Object {
    "opacity": 1.19,
  },
  "23%": Object {
    "opacity": 1.19,
  },
  "24%": Object {
    "opacity": 1.2,
  },
  "25%": Object {
    "opacity": 1.19,
  },
  "26%": Object {
    "opacity": 1.19,
  },
  "27%": Object {
    "opacity": 1.18,
  },
  "28%": Object {
    "opacity": 1.17,
  },
  "29%": Object {
    "opacity": 1.16,
  },
  "3%": Object {
    "opacity": 0.16,
  },
  "30%": Object {
    "opacity": 1.15,
  },
  "31%": Object {
    "opacity": 1.13,
  },
  "32%": Object {
    "opacity": 1.12,
  },
  "33%": Object {
    "opacity": 1.1,
  },
  "34%": Object {
    "opacity": 1.09,
  },
  "35%": Object {
    "opacity": 1.07,
  },
  "36%": Object {
    "opacity": 1.06,
  },
  "37%": Object {
    "opacity": 1.04,
  },
  "38%": Object {
    "opacity": 1.03,
  },
  "39%": Object {
    "opacity": 1.02,
  },
  "4%": Object {
    "opacity": 0.22,
  },
  "40%": Object {
    "opacity": 1.01,
  },
  "41%": Object {
    "opacity": 1,
  },
  "42%": Object {
    "opacity": 0.99,
  },
  "43%": Object {
    "opacity": 0.98,
  },
  "44%": Object {
    "opacity": 0.98,
  },
  "45%": Object {
    "opacity": 0.97,
  },
  "46%": Object {
    "opacity": 0.97,
  },
  "47%": Object {
    "opacity": 0.96,
  },
  "5%": Object {
    "opacity": 0.29,
  },
  "52%": Object {
    "opacity": 0.96,
  },
  "53%": Object {
    "opacity": 0.97,
  },
  "56%": Object {
    "opacity": 0.97,
  },
  "57%": Object {
    "opacity": 0.98,
  },
  "6%": Object {
    "opacity": 0.37,
  },
  "60%": Object {
    "opacity": 0.98,
  },
  "61%": Object {
    "opacity": 0.99,
  },
  "63%": Object {
    "opacity": 0.99,
  },
  "64%": Object {
    "opacity": 1,
  },
  "69%": Object {
    "opacity": 1,
  },
  "7%": Object {
    "opacity": 0.45,
  },
  "70%": Object {
    "opacity": 1.01,
  },
  "74%": Object {
    "opacity": 1.01,
  },
  "75%": Object {
    "opacity": 1,
  },
  "8%": Object {
    "opacity": 0.53,
  },
  "9%": Object {
    "opacity": 0.6,
  },
}
`;

exports[`css-spring matches snapshots 2`] = `
Object {
  "0%": Object {
    "background": "#bbd653",
    "padding-left": "-48px",
  },
  "1%": Object {
    "background": "#becf51",
    "padding-left": "-45px",
  },
  "10%": Object {
    "background": "#e9461b",
    "padding-left": "18px",
  },
  "100%": Object {
    "background": "#ff0000",
    "padding-left": "50px",
  },
  "11%": Object {
    "background": "#ee3615",
    "padding-left": "25px",
  },
  "12%": Object {
    "background": "#f3270f",
    "padding-left": "32px",
  },
  "13%": Object {
    "background": "#f71a0a",
    "padding-left": "38px",
  },
  "14%": Object {
    "background": "#fb0d05",
    "padding-left": "44px",
  },
  "15%": Object {
    "background": "#fe0201",
    "padding-left": "49px",
  },
  "16%": Object {
    "background": "#ff0000",
    "padding-left": "54px",
  },
  "17%": Object {
    "padding-left": "58px",
  },
  "18%": Object {
    "padding-left": "61px",
  },
  "19%": Object {
    "padding-left": "64px",
  },
  "2%": Object {
    "background": "#c1c54d",
    "padding-left": "-40px",
  },
  "20%": Object {
    "padding-left": "66px",
  },
  "21%": Object {
    "padding-left": "68px",
  },
  "22%": Object {
    "padding-left": "69px",
  },
  "23%": Object {
    "padding-left": "69px",
  },
  "24%": Object {
    "padding-left": "70px",
  },
  "25%": Object {
    "padding-left": "69px",
  },
  "26%": Object {
    "padding-left": "69px",
  },
  "27%": Object {
    "padding-left": "68px",
  },
  "28%": Object {
    "padding-left": "67px",
  },
  "29%": Object {
    "padding-left": "66px",
  },
  "3%": Object {
    "background": "#c5b848",
    "padding-left": "-34px",
  },
  "30%": Object {
    "padding-left": "65px",
  },
  "31%": Object {
    "padding-left": "63px",
  },
  "32%": Object {
    "padding-left": "62px",
  },
  "33%": Object {
    "padding-left": "60px",
  },
  "34%": Object {
    "padding-left": "59px",
  },
  "35%": Object {
    "padding-left": "57px",
  },
  "36%": Object {
    "padding-left": "56px",
  },
  "37%": Object {
    "padding-left": "54px",
  },
  "38%": Object {
    "padding-left": "53px",
  },
  "39%": Object {
    "padding-left": "52px",
  },
  "4%": Object {
    "background": "#c9aa42",
    "padding-left": "-28px",
  },
  "40%": Object {
    "background": "#ff0000",
    "padding-left": "51px",
  },
  "41%": Object {
    "background": "#ff0100",
    "padding-left": "50px",
  },
  "42%": Object {
    "background": "#fe0201",
    "padding-left": "49px",
  },
  "43%": Object {
    "background": "#fe0402",
    "padding-left": "48px",
  },
  "44%": Object {
    "background": "#fd0502",
    "padding-left": "48px",
  },
  "45%": Object {
    "background": "#fd0602",
    "padding-left": "47px",
  },
  "46%": Object {
    "background": "#fd0703",
    "padding-left": "47px",
  },
  "47%": Object {
    "background": "#fd0803",
    "padding-left": "46px",
  },
  "48%": Object {
    "background": "#fc0803",
  },
  "5%": Object {
    "background": "#ce9a3c",
    "padding-left": "-21px",
  },
  "52%": Object {
    "background": "#fc0803",
    "padding-left": "46px",
  },
  "53%": Object {
    "background": "#fd0803",
    "padding-left": "47px",
  },
  "54%": Object {
    "background": "#fd0703",
  },
  "55%": Object {
    "background": "#fd0703",
  },
  "56%": Object {
    "background": "#fd0602",
    "padding-left": "47px",
  },
  "57%": Object {
    "background": "#fd0502",
    "padding-left": "48px",
  },
  "58%": Object {
    "background": "#fe0502",
  },
  "59%": Object {
    "background": "#fe0402",
  },
  "6%": Object {
    "background": "#d38a36",
    "padding-left": "-13px",
  },
  "60%": Object {
    "background": "#fe0301",
    "padding-left": "48px",
  },
  "61%": Object {
    "background": "#fe0301",
    "padding-left": "49px",
  },
  "62%": Object {
    "background": "#fe0201",
  },
  "63%": Object {
    "background": "#ff0201",
    "padding-left": "49px",
  },
  "64%": Object {
    "background": "#ff0100",
    "padding-left": "50px",
  },
  "65%": Object {
    "background": "#ff0100",
  },
  "66%": Object {
    "background": "#ff0000",
  },
  "69%": Object {
    "padding-left": "50px",
  },
  "7%": Object {
    "background": "#d9792f",
    "padding-left": "-5px",
  },
  "70%": Object {
    "padding-left": "51px",
  },
  "8%": Object {
    "background": "#de6828",
    "padding-left": "3px",
  },
  "82%": Object {
    "padding-left": "51px",
  },
  "83%": Object {
    "padding-left": "50px",
  },
  "9%": Object {
    "background": "#e45722",
    "padding-left": "10px",
  },
}
`;

exports[`css-spring matches snapshots 3`] = `
Object {
  "0%": Object {
    "border": "1px solid #f00",
    "margin-right": "1.52em",
    "transform": "translateX(1px) translateY(1px) scaleX(1.01) scaleY(1.01)",
  },
  "1%": Object {
    "border": "1px solid #f00",
    "margin-right": "4.34em",
    "transform": "translateX(3px) translateY(3px) scaleX(1.03) scaleY(1.03)",
  },
  "10%": Object {
    "border": "5px solid #f00",
    "margin-right": "57.97em",
    "transform": "translateX(46px) translateY(46px) scaleX(1.46) scaleY(1.46)",
  },
  "100%": Object {
    "border": "10px solid #f00",
    "margin-right": "127.01em",
    "transform": "translateX(100px) translateY(100px) scaleX(2) scaleY(2)",
  },
  "11%": Object {
    "border": "6px solid #f00",
    "margin-right": "64.64em",
    "transform": "translateX(51px) translateY(51px) scaleX(1.51) scaleY(1.51)",
  },
  "12%": Object {
    "margin-right": "71.12em",
    "transform": "translateX(56px) translateY(56px) scaleX(1.56) scaleY(1.56)",
  },
  "13%": Object {
    "border": "6px solid #f00",
    "margin-right": "77.37em",
    "transform": "translateX(61px) translateY(61px) scaleX(1.61) scaleY(1.61)",
  },
  "14%": Object {
    "border": "7px solid #f00",
    "margin-right": "83.33em",
    "transform": "translateX(66px) translateY(66px) scaleX(1.66) scaleY(1.66)",
  },
  "15%": Object {
    "border": "7px solid #f00",
    "margin-right": "88.99em",
    "transform": "translateX(70px) translateY(70px) scaleX(1.7) scaleY(1.7)",
  },
  "16%": Object {
    "border": "8px solid #f00",
    "margin-right": "94.31em",
    "transform": "translateX(74px) translateY(74px) scaleX(1.74) scaleY(1.74)",
  },
  "17%": Object {
    "margin-right": "99.28em",
    "transform": "translateX(78px) translateY(78px) scaleX(1.78) scaleY(1.78)",
  },
  "18%": Object {
    "border": "8px solid #f00",
    "margin-right": "103.88em",
    "transform": "translateX(82px) translateY(82px) scaleX(1.82) scaleY(1.82)",
  },
  "19%": Object {
    "border": "9px solid #f00",
    "margin-right": "108.12em",
    "transform": "translateX(85px) translateY(85px) scaleX(1.85) scaleY(1.85)",
  },
  "2%": Object {
    "border": "2px solid #f00",
    "margin-right": "8.23em",
    "transform": "translateX(6px) translateY(6px) scaleX(1.06) scaleY(1.06)",
  },
  "20%": Object {
    "margin-right": "111.99em",
    "transform": "translateX(88px) translateY(88px) scaleX(1.88) scaleY(1.88)",
  },
  "21%": Object {
    "margin-right": "115.5em",
    "transform": "translateX(91px) translateY(91px) scaleX(1.91) scaleY(1.91)",
  },
  "22%": Object {
    "border": "9px solid #f00",
    "margin-right": "118.65em",
    "transform": "translateX(93px) translateY(93px) scaleX(1.93) scaleY(1.93)",
  },
  "23%": Object {
    "border": "10px solid #f00",
    "margin-right": "121.47em",
    "transform": "translateX(96px) translateY(96px) scaleX(1.96) scaleY(1.96)",
  },
  "24%": Object {
    "margin-right": "123.96em",
    "transform": "translateX(98px) translateY(98px) scaleX(1.98) scaleY(1.98)",
  },
  "25%": Object {
    "margin-right": "126.13em",
    "transform": "translateX(99px) translateY(99px) scaleX(1.99) scaleY(1.99)",
  },
  "26%": Object {
    "margin-right": "128.01em",
    "transform": "translateX(101px) translateY(101px) scaleX(2.01) scaleY(2.01)",
  },
  "27%": Object {
    "margin-right": "129.62em",
    "transform": "translateX(102px) translateY(102px) scaleX(2.02) scaleY(2.02)",
  },
  "28%": Object {
    "margin-right": "130.97em",
    "transform": "translateX(103px) translateY(103px) scaleX(2.03) scaleY(2.03)",
  },
  "29%": Object {
    "margin-right": "132.08em",
    "transform": "translateX(104px) translateY(104px) scaleX(2.04) scaleY(2.04)",
  },
  "3%": Object {
    "margin-right": "13.01em",
    "transform": "translateX(10px) translateY(10px) scaleX(1.1) scaleY(1.1)",
  },
  "30%": Object {
    "margin-right": "132.97em",
    "transform": "translateX(105px) translateY(105px) scaleX(2.05) scaleY(2.05)",
  },
  "31%": Object {
    "border": "10px solid #f00",
    "margin-right": "133.67em",
    "transform": "translateX(105px) translateY(105px) scaleX(2.05) scaleY(2.05)",
  },
  "32%": Object {
    "border": "11px solid #f00",
    "margin-right": "134.19em",
    "transform": "translateX(106px) translateY(106px) scaleX(2.06) scaleY(2.06)",
  },
  "33%": Object {
    "margin-right": "134.56em",
  },
  "34%": Object {
    "margin-right": "134.78em",
  },
  "35%": Object {
    "margin-right": "134.87em",
  },
  "36%": Object {
    "margin-right": "134.86em",
  },
  "37%": Object {
    "margin-right": "134.76em",
  },
  "38%": Object {
    "margin-right": "134.57em",
  },
  "39%": Object {
    "border": "11px solid #f00",
    "margin-right": "134.33em",
  },
  "4%": Object {
    "border": "2px solid #f00",
    "margin-right": "18.48em",
    "transform": "translateX(15px) translateY(15px) scaleX(1.15) scaleY(1.15)",
  },
  "40%": Object {
    "border": "10px solid #f00",
    "margin-right": "134.02em",
    "transform": "translateX(106px) translateY(106px) scaleX(2.06) scaleY(2.06)",
  },
  "41%": Object {
    "margin-right": "133.68em",
    "transform": "translateX(105px) translateY(105px) scaleX(2.05) scaleY(2.05)",
  },
  "42%": Object {
    "margin-right": "133.31em",
  },
  "43%": Object {
    "margin-right": "132.91em",
    "transform": "translateX(105px) translateY(105px) scaleX(2.05) scaleY(2.05)",
  },
  "44%": Object {
    "margin-right": "132.49em",
    "transform": "translateX(104px) translateY(104px) scaleX(2.04) scaleY(2.04)",
  },
  "45%": Object {
    "margin-right": "132.07em",
  },
  "46%": Object {
    "margin-right": "131.65em",
    "transform": "translateX(104px) translateY(104px) scaleX(2.04) scaleY(2.04)",
  },
  "47%": Object {
    "margin-right": "131.23em",
    "transform": "translateX(103px) translateY(103px) scaleX(2.03) scaleY(2.03)",
  },
  "48%": Object {
    "margin-right": "130.82em",
  },
  "49%": Object {
    "margin-right": "130.42em",
    "transform": "translateX(103px) translateY(103px) scaleX(2.03) scaleY(2.03)",
  },
  "5%": Object {
    "border": "3px solid #f00",
    "margin-right": "24.49em",
    "transform": "translateX(19px) translateY(19px) scaleX(1.19) scaleY(1.19)",
  },
  "50%": Object {
    "margin-right": "130.03em",
    "transform": "translateX(102px) translateY(102px) scaleX(2.02) scaleY(2.02)",
  },
  "51%": Object {
    "margin-right": "129.66em",
  },
  "52%": Object {
    "margin-right": "129.32em",
  },
  "53%": Object {
    "margin-right": "128.99em",
    "transform": "translateX(102px) translateY(102px) scaleX(2.02) scaleY(2.02)",
  },
  "54%": Object {
    "margin-right": "128.68em",
    "transform": "translateX(101px) translateY(101px) scaleX(2.01) scaleY(2.01)",
  },
  "55%": Object {
    "margin-right": "128.4em",
  },
  "56%": Object {
    "margin-right": "128.14em",
  },
  "57%": Object {
    "margin-right": "127.9em",
  },
  "58%": Object {
    "margin-right": "127.69em",
    "transform": "translateX(101px) translateY(101px) scaleX(2.01) scaleY(2.01)",
  },
  "59%": Object {
    "margin-right": "127.5em",
    "transform": "translateX(100px) translateY(100px) scaleX(2) scaleY(2)",
  },
  "6%": Object {
    "border": "3px solid #f00",
    "margin-right": "30.89em",
    "transform": "translateX(24px) translateY(24px) scaleX(1.24) scaleY(1.24)",
  },
  "60%": Object {
    "margin-right": "127.32em",
  },
  "61%": Object {
    "margin-right": "127.17em",
  },
  "62%": Object {
    "margin-right": "127.04em",
  },
  "63%": Object {
    "margin-right": "126.93em",
  },
  "64%": Object {
    "margin-right": "126.83em",
  },
  "65%": Object {
    "margin-right": "126.75em",
  },
  "66%": Object {
    "margin-right": "126.68em",
  },
  "67%": Object {
    "margin-right": "126.62em",
  },
  "68%": Object {
    "margin-right": "126.58em",
  },
  "69%": Object {
    "margin-right": "126.55em",
  },
  "7%": Object {
    "border": "4px solid #f00",
    "margin-right": "37.55em",
    "transform": "translateX(30px) translateY(30px) scaleX(1.3) scaleY(1.3)",
  },
  "70%": Object {
    "margin-right": "126.53em",
  },
  "71%": Object {
    "margin-right": "126.52em",
  },
  "72%": Object {
    "margin-right": "126.51em",
  },
  "73%": Object {
    "margin-right": "126.51em",
  },
  "74%": Object {
    "margin-right": "126.52em",
  },
  "75%": Object {
    "margin-right": "126.53em",
  },
  "76%": Object {
    "margin-right": "126.55em",
  },
  "77%": Object {
    "margin-right": "126.57em",
  },
  "78%": Object {
    "margin-right": "126.59em",
  },
  "79%": Object {
    "margin-right": "126.61em",
  },
  "8%": Object {
    "border": "4px solid #f00",
    "margin-right": "44.34em",
    "transform": "translateX(35px) translateY(35px) scaleX(1.35) scaleY(1.35)",
  },
  "80%": Object {
    "margin-right": "126.64em",
  },
  "81%": Object {
    "margin-right": "126.66em",
  },
  "82%": Object {
    "margin-right": "126.69em",
  },
  "83%": Object {
    "margin-right": "126.72em",
  },
  "84%": Object {
    "margin-right": "126.74em",
  },
  "85%": Object {
    "margin-right": "126.77em",
  },
  "86%": Object {
    "margin-right": "126.79em",
  },
  "87%": Object {
    "margin-right": "126.81em",
  },
  "88%": Object {
    "margin-right": "126.84em",
  },
  "89%": Object {
    "margin-right": "126.86em",
  },
  "9%": Object {
    "border": "5px solid #f00",
    "margin-right": "51.18em",
    "transform": "translateX(40px) translateY(40px) scaleX(1.4) scaleY(1.4)",
  },
  "90%": Object {
    "margin-right": "126.88em",
  },
  "91%": Object {
    "margin-right": "126.9em",
  },
  "92%": Object {
    "margin-right": "126.92em",
  },
  "93%": Object {
    "margin-right": "126.93em",
  },
  "94%": Object {
    "margin-right": "126.95em",
  },
  "95%": Object {
    "margin-right": "126.96em",
  },
  "96%": Object {
    "margin-right": "126.97em",
  },
  "97%": Object {
    "margin-right": "126.98em",
  },
  "98%": Object {
    "margin-right": "126.99em",
  },
  "99%": Object {
    "margin-right": "127em",
  },
}
`;
